axelrod/tests/unit/test_fingerprint.py

```diff
@@ -0,0 +1,133 @@
+        coordinates = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        self.assertEqual(test_coordinates, coordinates)
+
+    def test_create_probes(self):
+        coords = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        probes = create_probes(probe, coords)
+        self.assertEqual(len(probes), 4)
+        # self.assertEqual(probes, self.expected_probes)
+
+    def test_create_edges(self):
+        coords = [(0.0, 0.0), (0.0, 0.5), (0.5, 0.0), (0.5, 0.5)]
+        expected_edges = [(0, 2), (0, 3), (0, 4), (0, 5)]
+        edges = create_edges(coords)
+        self.assertEqual(edges, expected_edges)
+
+    def test_construct_tournament_elemets(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        edges, tournament_players = af.construct_tournament_elements(0.5)
+        self.assertEqual(edges, self.expected_edges)
+        self.assertEqual(len(tournament_players), 6)
+        self.assertEqual(tournament_players[0].__class__, af.strategy)
+
+    def test_fingerprint(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        af.fingerprint(turns=10, repetitions=2, step=0.25, processes=1)
+        edge_keys = sorted(list(af.interactions.keys()))
+        coord_keys = sorted(list(af.data.keys()))
+        self.assertEqual(af.step, 0.25)
+        self.assertEqual(edge_keys, sorted(long_edges))
+        self.assertEqual(coord_keys, long_coordinates)
+
+    def test_read_interactions(self):
+        filename = "test_outputs/test_results.csv"
+        af = AshlockFingerprint(self.strategy, self.probe)
+        edges, players = af.construct_tournament_elements(0.25)
+        spatial_tournament = axl.SpatialTournament(players, turns=10,
+                                                   repetitions=2,
+                                                   edges=edges)
+        spatial_tournament.play(build_results=False, filename=filename)
+        interactions = read_interactions(filename)
+        self.assertEqual(sorted(list(interactions.keys())), sorted(long_edges))
+
+    def test_generate_data(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        edges, players = af.construct_tournament_elements(0.25)
+        spatial_tournament = axl.SpatialTournament(players, turns=10,
+                                                   repetitions=2,
+                                                   edges=edges)
+        results = spatial_tournament.play(progress_bar=False,
+                                          keep_interactions=True)
+        data = generate_data(results.interactions, long_coordinates, edges)
+        keys = sorted(list(data.keys()))
+        values = [0 < score < 5 for score in data.values()]
+        self.assertEqual(sorted(keys), long_coordinates)
+        self.assertEqual(all(values), True)
+
+    def test_plot(self):
+        af = AshlockFingerprint(self.strategy, self.probe)
+        af.fingerprint(turns=10, repetitions=2, step=0.25, processes=1)
```

Comment:
I think you should run these without `processes` (when you're calling it with `processes=1`).